src/erlcloud_aws.erl

```diff
@@ -293,6 +290,15 @@ format_timestamp({{Yr, Mo, Da}, {H, M, S}}) ->
       io_lib:format("~4.10.0b-~2.10.0b-~2.10.0bT~2.10.0b:~2.10.0b:~2.10.0bZ",
                     [Yr, Mo, Da, H, M, S])).
 
+encode_params(Params, Headers) ->
+  LowerCaseHeaders = lists:map(fun({K, V}) -> {string:to_lower(K), V} end, Headers),
+  case proplists:get_value(<<"content-type">>, LowerCaseHeaders) of
```

Comment:
string vs binary?  how this can ever match?

Reply:
wow. not sure what I was thinking about. it should be just double quotes, right?

Reply:
yes.  the line just above does `string:to_lower(K)` and that is all strings

Reply:
sorry, it happened because in elixir string and binary are the same. completely forgot, that erlang strings are just lists of chars. Fixed